_posts/2017/2017-06-20-versioning-of-microservices.md

```diff
@@ -0,0 +1,75 @@
+---
+layout: post
+title: "Versioning of microservices"
+date: "2017-06-20 07:00:00"
+image: blog-header/microservices-pipeline-2.jpg
+categories: tech-stories
+authors: ["Benjamin", "Nicole"]
+---
+
+In the first post of our series "The Continuous Delivery Pipeline (CDP) of our Microservices Architecture" you learned about how we designed our continuous delivery pipeline to support our microservices architecture.
+Today it's all about versioning of microservices.
+
+## How do we keep track of the state and version of each microservice?
+
+Within our technical infrastructure the source code of each microservice is stored in a dedicated Git repository for version control.
+We use [GitHub](https://github.com/) to manage code contributions (e.g. to add new functionality or fix bugs), which are bundled in pull requests (PR).
+After passing a defined feedback cycle (automated tests and code analysis, developer code reviews, and quality assurance by a peer group) the PR is merged into the master branch, which then triggers a new build of the microservice.
+
+Up to this point there is nothing special about the code infrastructure management that we have in place.
+The interesting part happens in the build and publish job of each microservice when a new CDP run is triggered.
+
+We generate the build jobs for each microservice project (e.g. "microservice-alpha/beta/gamma") for each vertical (= bounded context in DDD in which fixed evolutionary multiple canonical domain models exist, e.g. "vertical-lambda") via the Jenkins Groovy DSL.
+
+{% highlight groovy %}
+def jobHelper = new helpers.JobHelper(this, System.env,'vertical-lambda');
+
+def projectNames = [
+    'microservice-alpha': ['triggerCDP': true],
+    'microservice-beta': ['triggerCDP': true],
+    'microservice-gamma': ['triggerCDP': true],
+]
+
+projectNames.keySet().each { projectName ->
+   jobHelper.gradleBuildAndPublishJob(projectName, projectNames[projectName]);
+   jobHelper.gradleVerifyPullRequestsJob(projectName);
+
```

Comment:
I had to remove "´´´" and "}" here, as the code wasn't highlighted correctly. I hope this example still makes sense? Please check.